x-pack/solutions/security/plugins/security_solution/public/siem_migrations/common/components/filters/status.tsx

```diff
@@ -10,110 +10,88 @@ import type { EuiSelectableOption } from '@elastic/eui';
 import { EuiFilterButton, EuiPopover, EuiSelectable } from '@elastic/eui';
 import type { EuiSelectableOnChangeEvent } from '@elastic/eui/src/components/selectable/selectable';
 
-import { MigrationTranslationResult } from '../../../../../common/siem_migrations/constants';
-import { StatusFilter } from '../../types';
-import { convertTranslationResultIntoText } from '../../utils';
 import * as i18n from './translations';
+import type { StatusFilterOptions } from './types';
 
 const STATUS_FILTER_POPOVER_WIDTH = 250;
 
-export interface StatusFilterButtonProps {
-  status?: StatusFilter;
-  onStatusChanged: (newStatus?: StatusFilter) => void;
+interface StatusFilterButtonProps<T> {
+  status?: T;
+  onStatusChanged: (newStatus?: T) => void;
+  statusFilterOptions: StatusFilterOptions<T>[];
 }
 
-export const StatusFilterButton: React.FC<StatusFilterButtonProps> = React.memo(
-  ({ status, onStatusChanged }) => {
-    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+const StatusFilterButtonComponent = <T,>({
+  status,
+  onStatusChanged,
+  statusFilterOptions,
+}: StatusFilterButtonProps<T>) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
 
-    const selectableOptions: EuiSelectableOption[] = [
-      {
-        label: i18n.INSTALL_FILTER_OPTION,
-        data: { status: StatusFilter.INSTALLED },
-        checked: status === StatusFilter.INSTALLED ? 'on' : undefined,
-      },
-      {
-        label: convertTranslationResultIntoText(MigrationTranslationResult.FULL),
-        data: { status: StatusFilter.TRANSLATED },
-        checked: status === StatusFilter.TRANSLATED ? 'on' : undefined,
-      },
-      {
-        label: convertTranslationResultIntoText(MigrationTranslationResult.PARTIAL),
-        data: { status: StatusFilter.PARTIALLY_TRANSLATED },
-        checked: status === StatusFilter.PARTIALLY_TRANSLATED ? 'on' : undefined,
-      },
-      {
-        label: convertTranslationResultIntoText(MigrationTranslationResult.UNTRANSLATABLE),
-        data: { status: StatusFilter.UNTRANSLATABLE },
-        checked: status === StatusFilter.UNTRANSLATABLE ? 'on' : undefined,
-      },
-      {
-        label: i18n.FAILED_FILTER_OPTION,
-        data: { status: StatusFilter.FAILED },
-        checked: status === StatusFilter.FAILED ? 'on' : undefined,
-      },
-      {
-        label: i18n.INDEX_PATTERN_MISSING_FILTER_OPTION,
-        data: { status: StatusFilter.INDEX_PATTERN_MISSING },
-        checked: status === StatusFilter.INDEX_PATTERN_MISSING ? 'on' : undefined,
-      },
-    ];
+  const selectableOptions: EuiSelectableOption[] = statusFilterOptions.map(({ label, data }) => ({
```

Comment:
`nit`: use `useMemo`? otherwise it will always be a new object.

Reply:
Done in https://github.com/elastic/kibana/pull/234486/commits/97b8d79be9124e47b060fee6dda205e8be2492d4